cylc/flow/task_proxy.py

```diff
@@ -297,6 +300,7 @@ def __init__(
             self.graph_children = generate_graph_children(tdef, self.point)
 
         self.mode_settings: Optional['ModeSettings'] = None
+        self.run_mode: Optional[Union[str, RunMode]] = None
```

Comment:
I think this should be either `str` or `RunMode`, rather than a union. Preferably `RunMode` for better type safety, though `str` would probably be easier to use than the Enum.